### ISSUE TITLE
Ambiguity warnings on master

### ISSUE THREAD

**Issue by jiahao** [JuliaStats/Distributions.jl#391]
Loading the master version of Distributions on julia 0.4 master now throws an ambiguity warning:

```
Warning: New definition 
    insupport(Distributions.Truncated{#D<:Distributions.Distribution{Distributions.Univariate, S<:Distributions.ValueSupport}, #S<:Distributions.ValueSupport}, Real) at /Users/jiahao/.julia/v0.4/Distributions/src/truncate.jl:33
is ambiguous with: 
    insupport(Union{Type{#D<:Distributions.Distribution{Distributions.Univariate, Distributions.Continuous}}, #D<:Distributions.Distribution{Distributions.Univariate, Distributions.Continuous}}, Real) at /Users/jiahao/.julia/v0.4/Distributions/src/univariates.jl:34.
To fix, define 
    insupport(_<:Distributions.Truncated{#D<:Distributions.Distribution{Distributions.Univariate, S<:Distributions.ValueSupport}, Distributions.Continuous}, Real)
before the new definition.
Warning: New definition 
    insupport(Distributions.Truncated{#D<:Distributions.Distribution{Distributions.Univariate, S<:Distributions.ValueSupport}, #S<:Distributions.ValueSupport}, Real) at /Users/jiahao/.julia/v0.4/Distributions/src/truncate.jl:33
is ambiguous with: 
    insupport(Union{Type{#D<:Distributions.Distribution{Distributions.Univariate, Distributions.Discrete}}, #D<:Distributions.Distribution{Distributions.Univariate, Distributions.Discrete}}, Real) at /Users/jiahao/.julia/v0.4/Distributions/src/univariates.jl:35.
To fix, define 
    insupport(_<:Distributions.Truncated{#D<:Distributions.Distribution{Distributions.Univariate, S<:Distributions.ValueSupport}, Distributions.Discrete}, Real)
before the new definition.
```

One of the set of ambiguous methods was introduced in #382; the other in #374.

cc: @brian-j-smith @simonbyrne - can we fix this?

**Comment by brian-j-smith** [comment]
I think this can be fixed by changing the ``insupport`` function in [truncated.jl](https://github.com/JuliaStats/Distributions.jl/blob/master/src/truncate.jl) to

```julia
insupport{D<:UnivariateDistribution}(d::Truncated{D,Union(Discrete,Continuous)}, x::Real) =
    d.lower <= x <= d.upper && insupport(d.untruncated, x)
```

Will do some testing tonight to make sure.

**Comment by brian-j-smith** [comment]
A proposed fix is in pull #392.

**Comment by jiahao** [comment]
Confirming fix. Thanks.
